Streetcode/Streetcode.BLL/Middleware/MiddlewareExtension.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Streetcode.BLL.Middleware
+{
+    public static class MiddlewareExtension
+    {
+        public static IApplicationBuilder UseMiddleware<T>(this IApplicationBuilder app )
+            where T : IMiddleware
+        {
+            return app.UseMiddleware(typeof(T));
+        }
+    }
```

Comment:
Why do you need this code?
You should add only extension to register RequestResponseMiddleware